packages/core/src/StateMachine.ts

```diff
@@ -442,9 +452,53 @@ export class StateMachine<
       State<TContext, TEvent, TResolvedTypesMeta>
     >
   ): State<TContext, TEvent, TResolvedTypesMeta> {
-    const restoredState = isStateConfig(state)
-      ? this.resolveState(state as any)
-      : this.resolveState(State.from(state as any, this.context, this));
+    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
+
+    if ('persisted' in state) {
+      const restoredChildren: Record<string, AnyActorRef> = {};
+
+      Object.keys(state.children).forEach((key) => {
+        const persistedState = state.children[key];
+        const impl = this.options.actors[key];
```

Comment:
this seems to be incorrect - `this.options.actors` is keyed using behavior keys~ but you are using an actor's id to index it here